src/ucs/sys/sock.h

```diff
@@ -184,6 +194,27 @@ ucs_status_t ucs_socket_send(int fd, const void *data, size_t length,
                              void *err_cb_arg);
 
 
+/**
+ * Non-blocking send operation sends I/O vector on the connected (or bound
+ * connectionless) socket referred to by the file descriptor `fd`.
+ *
+ * @param [in]      fd              Socket fd.
+ * @param [in]      iov             A pointer to an array of iovec buffers.
+ * @param [in]      iov_cnt         The number of buffers pointed to by
+ *                                  the iov parameter.
+ * @param [out]     length_p        The amount of data transmitted is written to
+ *                                  this argument.
+ * @param [in]      err_cb          Error callback.
+ * @param [in]      err_cb_arg      User's argument for the error callback.
+ *
+ * @return UCS_OK on success, UCS_ERR_CANCELED if connection closed,
```

Comment:
you do not mention `NO_PROGRESS`. I'd either mention all possible values or just state UCS_OK or error

Reply:
nice catch - done for `sendv_nb`, `send_nb` and `recv_nb`